### PR TITLE
Added Rundeck Chart

### PR DESCRIPTION
**Pull request by Deepak-Routray**
k8s Charts PR reference link https://github.com/kubernetes/charts/pull/6201

### CODE REVIEW

**Comment by dcwangmit01** [comment]
@Deepak-Routray 

Regarding the upstream PR, you should PR into the incubator directory instead of the stable directory.

File: pkg/index.yaml [hunk 1]
```diff
@@ -300,9 +300,22 @@ entries:
     urls:
     - https://raw.githubusercontent.com/cisco-sso/charts/master/pkg/nginx-0.4.2.tgz
     version: 0.4.2
+  rundeck:
+  - apiVersion: v1
+    appVersion: "1.0"
```
Comment (dcwangmit01) [inline pkg/index.yaml:305]: Should probably be semver.  e.g. 1.0.0

File: pkg/index.yaml [hunk 2]
```diff
@@ -300,9 +300,22 @@ entries:
...
+    created: 2018-06-20T10:48:05.24483179Z
```
Comment (dcwangmit01) [inline pkg/index.yaml:306]: MUST: Regarding the upstream pr, should go into incubator and not stable.  They will most def require you to do this.

File: src/rundeck/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Rundeck Helm chart
+This chart would install Rundeck in kubernetes cluster. Rundeck is open source software that helps automating routine operational procedures in data center or cloud environments. For more details on Rundeck, please refer [Rundeck](http://rundeck.org).
+Chart supports persistent shared store for storing rundeck project related files. It also supports external DB to store various jobs and their logs.
+## Prerequisite
+Kubernetes 1.9+
+Secret with name **rundeck-ssh** should exist in the k8s cluster which should use private key for remote edge nodes passwordless connection.
```
Comment (dcwangmit01) [inline src/rundeck/README.md:6]: MUST: This is not good.  Helm charts are supposed to work outof the box, without prereqs.

Please auto-gen the secret via a template and create in the k8s cluster.  Ensure that the secret is not clobbered upon helm upgrade.

File: src/rundeck/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Kubernetes 1.9+
+Secret with name **rundeck-ssh** should exist in the k8s cluster which should use private key for remote edge nodes passwordless connection.
+```
+kubectl create secret generic rundeck-ssh --from-file=id_rsa=<private key file name>
+```
+If external database is configured, then empty database should be available and DB related options inside values.yaml need to be configured accordingly.
+## Installing the chart
+
+```
+helm install --name rundeck .
```
Comment (dcwangmit01) [inline src/rundeck/README.md:14]: URL will change.   

helm install incubator/rundeck

File: src/rundeck/templates/rundeck-pv.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.volumeclaim.enabled}}
```
Comment (dcwangmit01) [inline src/rundeck/templates/rundeck-pv.yaml:1]: MUST: Don't use PVs.  Use StatefulSets

File: src/rundeck/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,115 @@
+{{/* store a reference '.' scope so Values can be access in side a loop
+  see https://github.com/kubernetes/helm/issues/1311 */}}
+{{- $root := . -}}
+
+apiVersion: apps/v1beta2
+kind: Deployment
```
Comment (dcwangmit01) [inline src/rundeck/templates/deployment.yaml:6]: MUST: Should be a statefulset instead.

File: src/rundeck/values.yaml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+resources:
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  limits:
+    cpu: 1
+    memory: 512Mi
+  requests:
+    cpu: 1
```
Comment (dcwangmit01) [inline src/rundeck/values.yaml:47]: SHOULD: Set to less by default.  100m

File: pkg/index.yaml (hunk 1 above)
Comment (ghost) [reply pkg/index.yaml:305]: Dave is right.  appVersion is not the chart version though, and must be changed in the chart yaml.

1.x is awhile ago in RunDeck releases http://rundeck.org/docs/history/changelog.html  @Deepak-Routray please correct it to match the version of RunDeck shipped in your container (not the container tag).

File: src/rundeck/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+name: rundeck
+version: 0.1.0
+home: https://github.com/Deepak-Routray/charts
+source:
+  - https://github.com/Deepak-Routray/rundeck-docker
+appVersion: "1.0"
+description: A Helm chart for Rundeck
+maintainers:
+  - name: DEEPAK ROUTRAY
+
```
Comment (ghost) [inline src/rundeck/Chart.yaml:11]: Please drop this empty line. the maintainer name should be your github username.

File: src/rundeck/templates/configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    
+    # ssh connection timeout after a specified number of milliseconds.
+    # "0" value means wait forever.
+    framework.ssh.timeout = 0
+    
+    
+    # ----------------------------------------------------------------
+    # Auto generated server UUID: 93be48ac-74f4-45f7-a809-bf6139d8a755
+    # ----------------------------------------------------------------
+    rundeck.server.uuid = 93be48ac-74f4-45f7-a809-bf6139d8a755
```
Comment (ghost) [inline src/rundeck/templates/configmap.yaml:54]: this is hardcoded, so how can it be auto generated?  Also, these config files should be in the values.yaml file and then included here in the config map. See other charts for examples of how this is done.

File: src/rundeck/templates/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+          - mountPath: {{ .Values.env.RSA_KEY_DIR }}/id_rsa
+            name: {{ template "rundeck.fullname" . }}-ssh-secret
+            subPath: id_rsa
+            readOnly: false          
+          {{- if .Values.storage.enabled}} 
+          - mountPath: {{ .Values.storage.mountPoint }}
+            name: rundeck-pvc
+            subPath: {{ .Values.storage.remotePath }}
+            readOnly: false
+          {{- end }}
```
Comment (ghost) [inline src/rundeck/templates/deployment.yaml:115]: No newline at end of this file. Please fix it so the red cross circle is gone.  https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: src/rundeck/templates/job-configmap.yaml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+metadata:
+  labels:
+    app: {{ template "rundeck.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    component: "{{ template "rundeck.name" . }}"
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "rundeck.fullname" . }}-job
+data:
+  project.properties: |-
```
Comment (ghost) [inline src/rundeck/templates/job-configmap.yaml:12]: Same as above: break files like this out into values.yaml.

File: src/rundeck/templates/rundeck-pvc.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.volumeclaim.enabled}}
```
Comment (ghost) [inline src/rundeck/templates/rundeck-pvc.yaml:1]: This must be dropped too in favor of the Statefulset pattern.

**Review by ghost — CHANGES_REQUESTED** [review]
@Deepak-Routray  I'm okay with you keeping the files in the configmaps and not migrating them to values.yaml for this first pass of the chart, but the PV / PVC thing and some other stuff must be fixed.

File: src/rundeck/templates/configmap.yaml (hunk 9 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/configmap.yaml:54]: Regarding moving config files in values.yaml. There are few variables which get repeated if we move this to values.yaml. E.g. env.SHARED_STORAGE and many others. That's the reason config are kept separately. If you prefer, we can use a hybrid mechanism for the same in next pass. Let me know your thoughts on this.

Comment (Deepak-Routray) [reply src/rundeck/templates/configmap.yaml:54]: uuid is auto generated now.

File: src/rundeck/templates/job-configmap.yaml (hunk 11 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/job-configmap.yaml:12]: This files has been removed as part of cleanup

File: src/rundeck/templates/rundeck-pvc.yaml (hunk 12 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/rundeck-pvc.yaml:1]: Dropped.

**Comment by Deepak-Routray** [comment]
@dcwangmit01 @josdotso 
By mistake, i created the PR from my forked master. As a result I had to delete the same.
Request you to keep this PR open for historical purpose.
I would create a new PR with all the feedback barring the few for which I provided the inline comments.
I would like to keep the chart into Stable in upstream k8s repo. Let us see how it goes.

File: src/rundeck/README.md (hunk 3 above)
Comment (ghost) [reply src/rundeck/README.md:6]: @Deepak-Routray  any update on this?

File: src/rundeck/README.md (hunk 4 above)
Comment (ghost) [reply src/rundeck/README.md:14]: @Deepak-Routray , and this?

File: src/rundeck/templates/configmap.yaml (hunk 9 above)
Comment (ghost) [reply src/rundeck/templates/configmap.yaml:54]: I don't see the code here for the auto generate @Deepak-Routray .. Link please?

File: src/rundeck/templates/configmap.yaml [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+    rundeck.tokens.file={{ .Values.env.SHARED_STORAGE }}/token/tokens.properties
+  realm.properties: |-
```
Comment (ghost) [inline src/rundeck/templates/configmap.yaml:56]: These config files should really be migrated into the values file.  See the stable/envoy chart for an example.

File: src/rundeck/templates/job-configmap.yaml (hunk 11 above)
Comment (ghost) [reply src/rundeck/templates/job-configmap.yaml:12]: link to code please. @Deepak-Routray ... I'm thinking it's probably in upstream, but please keep this PR in sync with what you push to upstream PR.  see the agreement.md in platform deploy.

**Review by ghost — CHANGES_REQUESTED** [review]
See comments. This PR is apparently out of sync with the upstream PR.

**Comment by Deepak-Routray** [comment]
closing this.

File: src/rundeck/README.md (hunk 3 above)
Comment (Deepak-Routray) [reply src/rundeck/README.md:6]: Yep. in New PR. Secret are not mandatory now.

File: src/rundeck/README.md (hunk 4 above)
Comment (Deepak-Routray) [reply src/rundeck/README.md:14]: Please refer new PR

File: src/rundeck/templates/configmap.yaml (hunk 9 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/configmap.yaml:54]: uuid generation is at dockerfile and shell script level. so not present in the chart

File: src/rundeck/templates/job-configmap.yaml (hunk 11 above)
Comment (Deepak-Routray) [reply src/rundeck/templates/job-configmap.yaml:12]: Refer new PR

**Pull request closed without merging**